laurent22/joplin
# Mobile: Plugin API: Add support for the renderMarkup command

# Summary

This pull request adds support for the `renderMarkup` command on mobile and web.

> [!NOTE]
>
> At present, the `renderMarkup` command runs in the main Hermes process and, as such, doesn't load user-installed plugins.
>

# Notes

- This changes the `renderMarkup` command to always return `file://` URLs, rather than `joplin-content://` URLs.
    - `joplin-content://` URLs don't seem to work from content within the toplevel frame (perhaps because it uses a `file://` URL?). This prevented images from rendering in the editor.
    - `joplin-content://` URLs don't work on Android/iOS.

# Testing plan

1. Build the work-in-progress version of the Extra Editor Settings plugin from [this branch](https://github.com/personalizedrefrigerator/joplin-plugin-extra-editor-settings/tree/wip/render-more-markup).
2. Add it to Joplin.
3. Set "Hide markup" to "More" in settings.
4. Open a note that contains images (with `![](:/...)`-style markup) and a Markdown table.
5. Verify that the images render in the editor when the cursor is on a different line.
   - **Note**: On web, images are not currently expected to render.
6. Verify that the table renders in the editor when the cursor is not near the table.

This has been tested successfully on:
- Desktop (Fedora 41)
- Web (Chromium 131)
- Android (API 35)
- iOS 18.2.

Additional regression testing (desktop only):
1. Open a Kanban board (using the YesYouKan plugin).
2. Add an image, using the Markdown editor, to one of the cards.
3. Open the Kanban editor.
4. Verify that the image renders.
